### PR TITLE
test: translate simple tests to DSL

### PR DESCRIPTION
**Pull request by strib**
In preparation for an upcoming PR that changes the way syncing
directory operations works, which would break all of the mock tests

Issue: KBFS-2076

fixup

### CODE REVIEW

**Comment by songgao** [comment]
If I understand correctly, the added simple tests aren't equivalent to all of the removed tests. Is this intentional? (to remove some unnecessary tests)

**Comment by strib** [comment]
@songgao: my intention was to keep the same coverage.  Some of the tests (e.g., `TestKBFSOpRemoveMultiBlockFileSuccess`) already had equivalents in the DSL world, so I didn't need to make new tests for them.  If you find other specific tests that aren't covered under DSL, please let me know!

**Comment by songgao** [comment]
Gocha. Thanks!

File: test/simple_test.go [hunk 1]
```diff
@@ -11,16 +11,427 @@ import (
...
+func TestCreateDirInRoot(t *testing.T) {
+	test(t,
+		users("alice", "bob"),
+		as(alice,
+			mkdir("a"),
+			// Initial check before SyncAll is called.
+			lsdir("", m{"a$": "DIR"}),
+			lsdir("a", m{}),
+		),
+		as(alice,
```
Comment (jzila) [inline test/simple_test.go]: Is this implicitly after the `SyncAll`?

Comment (strib) [inline test/simple_test.go]: Yes, the `SyncAll` happens at the end of the previous `as` block.

**Comment by strib** [comment]
> out of curiosity, what's the test coverage before and after this change? (go test -cover, I think)

@akalin-keybase:

master:
```
kbfs$ go test -cover ./libkbfs ./libfuse ./test 2>&1 | tee /tmp/cover-master.out
ok  	github.com/keybase/kbfs/libkbfs	11.943s	coverage: 66.6% of statements
ok  	github.com/keybase/kbfs/libfuse	5.276s	coverage: 49.7% of statements
ok  	github.com/keybase/kbfs/test	36.368s	coverage: 86.2% of statements
```

This PR:
```
kbfs$ go test -cover ./libkbfs ./libfuse ./test 2>&1 | tee /tmp/cover-pr1.out
ok  	github.com/keybase/kbfs/libkbfs	10.579s	coverage: 66.4% of statements
ok  	github.com/keybase/kbfs/libfuse	5.382s	coverage: 49.7% of statements
ok  	github.com/keybase/kbfs/test	34.776s	coverage: 86.2% of statements
```

So pretty negligible effect.  I don't think `-cover` counts coverage between packages, unfortunately.

**Pull request merged**